lib/internal/crypto/keygen.js

```diff
@@ -245,6 +247,47 @@ function check(type, options, callback) {
                                             cipher, passphrase, wrap);
       }
       break;
+    case 'dh':
+      {
+        const { group, primeLength, prime, generator } = needOptions();
+        let args;
+        if (group != null) {
+          let incompatibleOption;
+          if (((incompatibleOption = 'prime') && prime != null) ||
+              ((incompatibleOption = 'primeLength') && primeLength != null) ||
+              ((incompatibleOption = 'generator') && generator != null))
+            throw new ERR_INCOMPATIBLE_OPTION_PAIR('group', incompatibleOption);
+          if (typeof group !== 'string')
+            throw new ERR_INVALID_OPT_VALUE('group', group);
+          args = [group];
+        } else {
+          if (prime != null) {
+            if (primeLength != null)
+              throw new ERR_INCOMPATIBLE_OPTION_PAIR('prime', 'primeLength');
+            if (!isArrayBufferView(prime))
+              throw new ERR_INVALID_OPT_VALUE('prime', prime);
```

Comment:
Nit: `validateBuffer()`

Reply:
Same as above:

> I don't think throwing `ERR_INVALID_ARG_TYPE` is appropriate for options. Or if it is, why do we have `ERR_INVALID_OPT_VALUE`?

Reply:
It is possible to use it as: `validateBuffer(prime, 'options.prime');`. That way the error code is still `ERR_INVALID_ARG_TYPE` but the error message is appropriate for properties.